### PR TITLE
Expose libutils APIs that should have been public

### LINKED ISSUE #2265: Unexposed library are using in Samples
**Describe the bug**
I'm wondering why there's some library that doesn't expose in public install library but using in Samples (Such like NameComponentManager). I thought NameComponentManager should be public and it's also a sample for using SingleInstanceComponentManager.h. Or are those two files meant to be hidden?

I can't find a way to load out the node name from gltf file using AssetLoader with default filament public install library.

### PR DESCRIPTION
**Pull request by romainguy**
Expose APIs used by other libraries' public headers and by sample apps.

This PR fixes #2265.

### CODE REVIEW

File: libs/utils/include/utils/JobSystem.h [hunk 1]
```diff
@@ -36,7 +37,7 @@
 
 namespace utils {
 
-class JobSystem {
+class UTILS_PUBLIC JobSystem {
```
Comment (pixelflinger) [inline libs/utils/include/utils/JobSystem.h]: We sure about this one?

Comment (romainguy) [reply libs/utils/include/utils/JobSystem.h]: Removed

**Comment by bejado** [comment]
`SingleInstanceComponentManager.h` needs `<tsl/robin_map.h>`.. is that public too?

**Comment by romainguy** [comment]
It should be a single header.

**Comment by prideout** [comment]
LGTM

**Pull request merged**